### PR TITLE
Do not crash on tid == -1 in sys_thr_set_name

### PR DESCRIPTION
**Pull request by mikusp**
Previous implementation thought of tid == -1 but it crashed on unwrapping the conversion to i32. This is called by system app ShellAppProxy

### CODE REVIEW

File: src/kernel/src/process/mod.rs [hunk 1]
```diff
@@ -421,7 +421,7 @@ impl VProc {
     }
 
     fn sys_thr_set_name(self: &Arc<Self>, i: &SysIn) -> Result<SysOut, SysErr> {
-        let tid: i32 = i.args[0].try_into().unwrap();
+        let tid: i64 = i.args[0].try_into().unwrap();
```
Comment (SuchAFuriousDeath) [inline src/kernel/src/process/mod.rs:424]: This is an infallible conversion, it should be .into(). You'll have to write a From implementation for this I think

**Review by SuchAFuriousDeath — DISMISSED** [review]

File: src/kernel/src/process/mod.rs (hunk 1 above)
Comment (mikusp) [reply src/kernel/src/process/mod.rs]: There already is a From instance for that 👍

Comment (SuchAFuriousDeath) [reply src/kernel/src/process/mod.rs]: Oh right, I actually wrote it, but I thought it was in one of the yet unmerged PRs :D

**Review by SuchAFuriousDeath — APPROVED** [review]

**Review by VocalFan — APPROVED** [review]

**Comment by SuchAFuriousDeath** [comment]
I told VocalFan to merge this, because it's already done this way in sys_cpuset_getaffinity

**Pull request merged**